casper/src/main/resources/PoS.rhox

```diff
@@ -246,21 +315,26 @@ new  PoS,
             } |
 
             contract PoS(@"closeBlock", ackCh) = {
-              new commitRewards,
+              new accumulateDeposits,
+                  commitRewards,
                   changeEpoch,
                   removeQuarantinedWithdrawers,
                   payValidators
               in {
                 new blockDataCh, mvarProcessCh,
                     getBlockData(`rho:block:data`),
+                    accDepositsDoneCh,
                     commitRewardsCh,
                     newValidatorsCh, removeQuarantinedCh, paymentDoneCh
                 in {
                   getBlockData!(*blockDataCh) |
 
                   runMVar!(*stateCh, *mvarProcessCh, *ackCh) |
-                  for(_, @blockNumber <- blockDataCh;
+                  for(_, @blockNumber, _ <- blockDataCh;
                       @state, mvarResultCh <- mvarProcessCh) {
+                    // Before choosing new active validators and paying out any withdrawers,
+                    // deposits have to be accumulated into one vault for now.
+                    accumulateDeposits!(posRevAddress, *accDepositsDoneCh) |
```

Comment:
I want to state the logic here to make sure I understand correctly. 
Transfer all the all the bonding deposit to a `posRevAddress` and when a validator is unbonding the node, the `posRevAddress` put back the stake

Reply:
Yes, that sounds correct.